packages/iTwinUI-react/src/core/Table/hooks/useScrollToRow.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import type { Row } from 'react-table';
+import { TableProps } from '../Table';
+
+type ScrollToRow<T extends Record<string, unknown>> = {
+  scrollToIndex: number | undefined;
+  tableRowRef: (row: Row<T>) => (element: HTMLDivElement) => void;
+};
+
+type ScrollToRowProps<T extends Record<string, unknown>> = Pick<
+  TableProps<T>,
+  'enableVirtualization' | 'getRowId' | 'paginatorRenderer'
+> & { page: Row<T>[]; scrollToItem?: T };
+
+export function useScrollToRow<T extends Record<string, unknown>>({
+  enableVirtualization,
+  scrollToItem,
+  page,
+  getRowId,
+  paginatorRenderer,
+}: ScrollToRowProps<T>): ScrollToRow<T> {
+  const rowRefs = React.useRef<Record<string, HTMLDivElement>>({});
+
+  // For virtualized tables, all we need to do is pass the index of the item
+  // to the VirtualScroll component
+  const scrollToIndex = React.useMemo((): number | undefined => {
+    if (
+      !enableVirtualization ||
+      !getRowId ||
+      !scrollToItem ||
+      paginatorRenderer
+    ) {
+      return undefined;
+    }
+
+    const index = page.findIndex(
+      (row) => getRowId(row.original, 0) == getRowId(scrollToItem, 0),
```

Comment:
Ugh, I started to think that this might cause us problems because we are not calling `getRowId` as it is intended because first parameter is row data, second is row index and third is row parent if table has subrows. So, if someone uses all these 3 parameters in their `getRowId`, we are screwed. This is how default `getRowId` looks like:
![image](https://user-images.githubusercontent.com/36186912/177746964-8a862b94-2abb-4388-9010-92921aee6838.png)
What if we make `scrollToItem?: (rows: Rows<T>) => Row<T>`. Then users could use however they want e.g.
 `scrollToItem={(rows) => rows.find(row => row.id === item.id)}` 
or if `data` is not being changed
`scrollToItem={(rows) => rows.find(row => row.original === item)}`

Reply:
Hey, I started to work on this, in case you won't implement the same

Reply:
I tested this out in my app, both virtualized and non-virtualized, sorted and filtered. Seems to be working just a-ok! Thanks! 👍